optimum/habana/transformers/trainer.py

```diff
@@ -673,6 +673,10 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args):
                 transformers.modeling_utils.checkpoint = lazy_mode_checkpointing
 
             self.model.gradient_checkpointing_enable(gradient_checkpointing_kwargs=gradient_checkpointing_kwargs)
+
+            # Wrap `_gradient_checkpointing_func` in the model with `transformer_engine` `activation_checkpointing` context.
```

Comment:
Add short line on what is activation_checkpointing context and why is it needed here.

Reply:
Done